sql-serializer/src/test/java/io/cdap/pipeline/sql/serializer/BigQuerySerializerTest.java

```diff
@@ -0,0 +1,130 @@
+    .select(VALID_NO_TABLE_ALIAS_COL1, VALID_NO_TABLE_ALIAS_COL2)
+    .from(VALID_TABLE2).where(NO_TABLE_ALIAS_FILTER).as("t2").build();
+  private static final Column NESTED_NO_TABLE_ALIAS_COL1 = new Column("c1", VALID_QUERY_WITH_ALIAS, "cn1");
+  private static final Column NESTED_NO_TABLE_ALIAS_COL2 = new Column("col2", VALID_QUERY_WITH_ALIAS, null);
+  private static final Filter NO_TABLE_ALIAS_SUBFILTER1 =
+    new Filter(NESTED_NO_TABLE_ALIAS_COL1, STRING_CONSTANT2, PredicateOperatorType.EQUAL);
+  private static final Filter NO_TABLE_ALIAS_SUBFILTER2 =
+    new Filter(NESTED_NO_TABLE_ALIAS_COL2, INTEGER_CONSTANT, PredicateOperatorType.GREATER_OR_EQUAL);
+  private static final Filter NO_TABLE_ALIAS_NESTED_FILTER =
+    new Filter(NO_TABLE_ALIAS_SUBFILTER1, NO_TABLE_ALIAS_SUBFILTER2, PredicateOperatorType.OR);
+  private static final StructuredQuery VALID_NESTED_NO_TABLE_ALIAS = StructuredQuery.builder()
+    .select(NESTED_NO_TABLE_ALIAS_COL1, NESTED_NO_TABLE_ALIAS_COL2)
+    .from(VALID_NO_TABLE_ALIAS_QUERY_WITH_ALIAS).where(NO_TABLE_ALIAS_NESTED_FILTER).build();
+
+  // valid temporary table
+  private static final Table VALID_TEMPORARY_TABLE = new Table(null, null, "t1", null);
+
+  // result strings
+  private static final String EXPECTED_SIMPLE_QUERY_STRING =
+    "CREATE TEMPORARY TABLE `t1` AS SELECT d.col1 AS c1, d.col2 FROM `a.b.c` AS d WHERE (d.col1 = \"test1\");";
+  private static final String EXPECTED_NESTED_QUERY_STRING =
+    "CREATE TEMPORARY TABLE `t1` AS SELECT t2.c1 AS cn1, t2.col2 FROM (SELECT d.col1 AS c1, d.col2 FROM `a.b.c` " +
+      "AS d WHERE (d.col1 = \"test1\")) AS t2 WHERE ((t2.c1 = \"test2\") OR (t2.col2 >= 15));";
+  private static final String EXPECTED_NO_TABLE_ALIAS_QUERY_STRING =
+    "CREATE TEMPORARY TABLE `t1` AS SELECT t2.c1 AS cn1, t2.col2 FROM (SELECT col1 AS c1, col2 FROM `a.b.c` " +
+    "WHERE (col1 = \"test1\")) AS t2 WHERE ((t2.c1 = \"test2\") OR (t2.col2 >= 15));";
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testNullQuery() {
+    Table temp = new Table(null, null, "a", null);
+    BigQuerySQLSerializer serializer = new BigQuerySQLSerializer();
+    serializer.getSQL(null, temp);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testInvalidQualifiedTemporaryTable() {
+    BigQuerySQLSerializer serializer = new BigQuerySQLSerializer();
+    Table temp = new Table("a", null, "b", null);
+    serializer.getSQL(VALID_QUERY, temp);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testInvalidAliasedTemporaryTable() {
+    BigQuerySQLSerializer serializer = new BigQuerySQLSerializer();
+    Table temp = new Table(null, null, "b", "c");
+    serializer.getSQL(VALID_QUERY, temp);
+  }
+
+  @Test
+  public void testSimpleQuery() {
+    BigQuerySQLSerializer serializer = new BigQuerySQLSerializer();
+    Assert.assertEquals(EXPECTED_SIMPLE_QUERY_STRING, serializer.getSQL(VALID_QUERY, VALID_TEMPORARY_TABLE));
```

Comment:
For tests, it's more important for things to be readable than to avoid hardcoding strings and re-using variables. Somebody unfamiliar with the code should be able to look at the method and understand what it is testing without hopping back and forth in the file.

If I just look at the constants defined above, I don't know which ones are used where. Can you move the constants into local variables in the individual test methods.

Reply:
Fixed by moving all of the instance variables to inside of the method calls.